src/video/sdl2_v.cpp

```diff
@@ -54,19 +56,12 @@ static SDL_Palette *_sdl_palette;
 static bool _cursor_new_in_window = false;
 #endif
 
-#define MAX_DIRTY_RECTS 100
-static SDL_Rect _dirty_rects[MAX_DIRTY_RECTS];
-static int _num_dirty_rects;
+static Rect _dirty_rect;
 
 void VideoDriver_SDL::MakeDirty(int left, int top, int width, int height)
 {
-	if (_num_dirty_rects < MAX_DIRTY_RECTS) {
-		_dirty_rects[_num_dirty_rects].x = left;
-		_dirty_rects[_num_dirty_rects].y = top;
-		_dirty_rects[_num_dirty_rects].w = width;
-		_dirty_rects[_num_dirty_rects].h = height;
-	}
-	_num_dirty_rects++;
+	Rect r = {left, top, left + width, top + height};
+	_dirty_rect = BoundingRect(_dirty_rect, r);
```

Comment:
You may have to add the "_dirty_rect.bottom == 0" magic here as well.

Reply:
Though I would prefer a "bool any_dirty"

Reply:
what magic do you refer to? This function to me seems to be doing exactly what I would like? Make the biggest rect in which the old and new rect fit?

Reply:
Ah, you mean for the first call here, where `_dirty_rect` is all zeros. This would mean `top` and `left` always remain zero. As I copied this from the OpenGL branch, lets ask the wizard that wrote it what to do: fix it in `MakeDirty` or fix it in `BoundingRect` :D